editor/plugins/script_text_editor.cpp

```diff
@@ -1726,7 +1726,19 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 
 				String variable_name = String(node->get_name()).to_snake_case().validate_identifier();
 				if (use_type) {
-					text_to_drop += vformat("@onready var %s: %s = %s%s\n", variable_name, node->get_class_name(), is_unique ? "%" : "$", path);
+					String class_name = node->get_class_name();
+					Ref<Script> node_script = node->get_script_instance()->get_script();
+					if (node_script != nullptr) {
+						String script_path = node_script->get_path();
+						TypedArray<Dictionary> class_list = ProjectSettings::get_singleton()->get_global_class_list();
+						for (int index = 0; index < class_list.size(); index++) {
+							if (class_list[index].get("path") == script_path) {
+								class_name = class_list[index].get("class");
+								break;
+							}
+						}
+					}
```

Comment:
You can use `script->get_global_name()`

Reply:
I've been doing that in GDScript many times, and you had a better way o,.,o